File: medpay-web/src/types/prescription.ts
export interface PrescriptionItemRequest {
  productId: string;
  quantity: number;
  dosageInstruction?: string;
  frequency?: string;
  durationDays?: number;
}

export interface PrescriptionCreateRequest {
  patientId: string;
  appointmentId?: string;
  diagnosis: string;
  diagnosisCode?: string;
  notes?: string;
  items: PrescriptionItemRequest[];
}

export interface PrescriptionItemResponse {
  id: string;
  productId: string;
  productName: string;
  specification: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
  dosageInstruction: string;
  frequency: string;
  durationDays: number | null;
  insuranceCovered: boolean;
  insuranceRatio: number;
}

export interface PrescriptionResponse {
  id: string;
  prescriptionNo: string;
  doctorId: string;
  patientId: string;
  appointmentId: string | null;
  diagnosis: string;
  diagnosisCode: string | null;
  notes: string | null;
  totalAmount: number;
  insuranceAmount: number;
  selfPayAmount: number;
  status: string;
  validUntil: string;
  createdAt: string;
  items: PrescriptionItemResponse[];
}

File: medpay-catalog/src/main/java/com/medpay/catalog/domain/Prescription.java
package com.medpay.catalog.domain;

import com.medpay.common.domain.TenantEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "prescription")
public class Prescription extends TenantEntity {

    @Column(name = "prescription_no", unique = true, nullable = false)
    private String prescriptionNo;

    @Column(name = "doctor_id", nullable = false, columnDefinition = "uuid")
    private UUID doctorId;

    @Column(name = "patient_id", nullable = false, columnDefinition = "uuid")
    private UUID patientId;

    @Column(name = "appointment_id", columnDefinition = "uuid")
    private UUID appointmentId;

    @Column(name = "diagnosis")
    private String diagnosis;

    @Column(name = "diagnosis_code")
    private String diagnosisCode;

    @Column(name = "notes")
    private String notes;

    @Column(name = "total_amount")
    private BigDecimal totalAmount;

    @Column(name = "insurance_amount")
    private BigDecimal insuranceAmount;

    @Column(name = "self_pay_amount")
    private BigDecimal selfPayAmount;

    @Column(name = "status")
    private String status = "PENDING";

    @Column(name = "valid_until")
    private LocalDateTime validUntil;
}
